packages/agoric-cli/src/deploy.js

```diff
@@ -89,9 +103,60 @@ export default async function deployMain(progname, rawArgs, powers, opts) {
     1000,
   );
 
-  const retryWebsocket = async () => {
-    const accessToken = await getAccessToken(opts.hostport);
+  const accessToken = await getAccessToken(opts.hostport);
 
+  const jsonHttpCall = makeJsonHttpClient({ http });
+
+  const listConnections = async () => {
+    const { ok, connections } = await jsonHttpCall({
+      hostname: host,
+      port,
+      method: 'GET',
+      path: `/connections?accessToken=${encodeURIComponent(accessToken)}`,
+    });
+    assert(
+      ok === true,
+      `Expected JSON body "ok" property to be true for HTTP request for chain connections`,
+    );
+    return connections;
+  };
+
+  const getDefaultConnection = async () => {
+    const connections = await listConnections();
+    console.log('\nConnection options:');
+    const lookup = new Map(
+      connections.map((connection, index) => {
+        console.log(`${index + 1}.`, connection);
+        return [`${index + 1}`, connection];
+      }),
+    );
+    const { connectionNumber } = await inquirer.prompt({
+      name: 'connectionNumber',
+      message: 'Please choose a connection to publish to',
+      choices: lookup.keys(),
+    });
+    const connection = lookup.get(connectionNumber);
+    console.log({ connection });
+    return connection;
+  };
+
+  const pspawnEnv = { ...process.env, DEBUG: 'agoric,deploy,deploy:publish' };
+  const pspawn = makePspawn({ env: pspawnEnv, spawn, log: console, chalk });
+  const { cosmosHelper } = getSDKBinaries(sdkPrefixes);
+  const publishBundleCosmos = makeCosmosBundlePublisher({
+    pspawn,
+    cosmosHelper,
+    pathResolve: path.resolve,
+    writeFile: fs.writeFile,
+  });
+  const publishBundle = makeBundlePublisher({
+    getAccessToken,
+    jsonHttpCall,
+    publishBundleCosmos,
+    getDefaultConnection,
+  });
```

Comment:
I’m a little dissatisfied that the HTTP guts are inside `@agoric/publish-bundle` but the Cosmos SDK guts are in `@agoric/agoric`. Putting it all here makes it harder to test in `@agoric/solo`, which doesn’t work in `solo/test/test-home.js` yet anyway. Putting all the guts in `@agoric/publish-bundle` would require some duplication or refactoring of the utilities above that are needed to make the bundle publisher.